src/ex1/script.js

```diff
@@ -0,0 +1,165 @@
+addTodoForm.addEventListener('submit', onAddTodoFormSubmitted);
+sortListButton.addEventListener('click', onSortListButtonClicked);
+
+todosArray.forEach(todoText => {
+  addTodoItem(todoText, false);
+  displayFooterAndImage();
+});
+
+function addEventListenerForTodoTitle(listItem) {
+  const todoTitleDiv = listItem.getElementsByTagName("div")[0];
+  todoTitleDiv.addEventListener('click', ({target}) => {
+    onTodoTitleClicked(target);
+  });
+}
+
+function addEventListenerForDeleteButton(listItem) {
+  const deleteButton = listItem.getElementsByTagName("button")[0]
+  deleteButton.addEventListener('click', ({currentTarget}) => {
+    onDeleteButtonClicked(currentTarget);
+  });
+}
+
+function addTodoItem(todoText, animation) {
+  const listItem = createListElement(todoText, animation);
+  addEventListenerForTodoTitle(listItem);
+  addEventListenerForDeleteButton(listItem);
+  allTodosList.appendChild(listItem);
+  isListSorted = unsorted;
+}
+
+function createListElement(todoText, animation) {
+  const listItem = document.createElement("li");
+  if (animation) {
+    listItem.className = "add-item-animation";
+  }
+  setTimeout (() => { listItem.className = "existing-item";}, 700);
+  listItem.innerHTML = `<div class="todo-text">${todoText}</div>
+                        <button class="remove-todo-button"><i class="fa fa-trash"></i></button>`;
+  return listItem;
+}
+
+function displayFooterAndImage() {
+  displayButtonsAndAmount();
+  displayZeroImage();
+}
+
+function displayButtonsAndAmount() {
+  let tasks = "tasks";
+  if (todosArray.length === 1) {
+    tasks = "task";
+    displayElement('sort-list-button', false)
+    displayElement('clear-all-button', false)
+  } else {
+    displayElement('sort-list-button', true)
+    displayElement('clear-all-button', true)
+  }
+  amountOfTodosInfo.textContent = `${todosArray.length} pending ${tasks}`;
+}
+
+function displayZeroImage() {
+  if (todosArray.length === 0) {
+    displayElement('zero-todos-image', true)
+    displayElement('footer', false)
+  } else {
+    displayElement('zero-todos-image', false)
+    displayElement('footer', true)
+  }
+}
+
+function displayElement(elementId, toDisplay) {
+  displayStyle = toDisplay ? "" : "none";
+  document.getElementById(elementId).style.display = displayStyle;
+}
+
+function onTodoTitleClicked(clickedTodo) {
+  alert(clickedTodo.textContent);
+}
+
+function onDeleteButtonClicked(clickedButton) {
+  const index = Array.prototype.indexOf.call(allTodosList.getElementsByTagName("li"), clickedButton.parentElement);
+  todosArray.splice(index, 1);
+  clickedButton.parentElement.classList.remove("existing-item");
+  clickedButton.parentElement.classList.add("delete-item-animation");
+  setTimeout (() => {
+    clickedButton.parentElement.remove();
+    displayFooterAndImage();
+  }, 700);
+}
+
+function onClearAllButtonClicked() {
+  const deleteButtons = Array.prototype.slice.call(document.querySelectorAll("div"));
```

Comment:
very dangerous selector "div" as any addition to the HTML code and usage of div will result in a bug here, prefer using more specific selectors

Reply:
fixed